css/src/components/checkbox.scss

```diff
@@ -0,0 +1,118 @@
+/* stylelint-disable selector-max-specificity, max-nesting-depth, selector-no-qualifying-type */
+
+$checkbox-border-radius: $radius-sm !default;
+$checkbox-border-color: $text !default;
+$checkbox-check-hover-color: $secondary !default;
+$checkbox-icon-size: 0.875rem !default;
+$checkbox-size: 1.25rem !default;
+$checkbox-size-small: 0.875rem !default;
+$checkbox-icon-size: 1rem !default;
+$checkbox-icon-size-small: 0.625rem !default;
+$checkbox-timing-function: $input-timing-function !default;
+$checkbox-duration: $input-transition-duration !default;
+$checkbox-spacing: $spacer-3;
+
+.checkbox {
+	display: flex;
+	position: relative;
+	align-items: center;
+	line-height: 1.25;
+	cursor: pointer;
+
+	+ .checkbox {
+		margin-top: $checkbox-spacing;
+		margin-left: 0;
+	}
+
+	.checkbox-check {
+		display: flex;
+		flex-shrink: 0;
+		align-items: center;
+		justify-content: center;
+		width: $checkbox-size;
+		height: $checkbox-size;
+		transition-duration: $checkbox-duration;
+		transition-property: background, border, border-color;
+		transition-timing-function: $checkbox-timing-function;
+		border: 1px solid $checkbox-border-color;
+		border-radius: $checkbox-border-radius;
+
+		&:not(:last-child) {
+			margin-#{$user-right}: 0.5rem;
+		}
+
+		&::before {
+			font-size: $checkbox-icon-size;
+			content: '\0020';
+		}
+	}
+
+	&:hover {
+		.checkbox-check::before {
+			@include checkmark;
+		}
+	}
+
+	&.is-small {
+		.checkbox-check {
+			width: $checkbox-size-small;
+			height: $checkbox-size-small;
+
+			&::before {
+				top: 2px !important;
+				width: calc($checkbox-icon-size-small / 2) !important;
+				height: $checkbox-icon-size-small !important;
+			}
+		}
+
+		.checkbox-text {
+			font-size: $checkbox-size-small;
+		}
+	}
+
+	input[type='checkbox'] {
+		@include visually-hidden();
+
+		&:focus-visible + .checkbox-check {
+			@extend %focus;
+		}
+
+		&:checked + .checkbox-check {
+			border-color: $primary;
+			background-color: $primary;
+
+			&::before {
+				@include checkmark;
+
+				border-color: $primary-invert;
+			}
+		}
+
+		&[disabled] + .checkbox-check {
+			opacity: 0.6;
+			cursor: not-allowed;
+		}
+	}
+
+	&.is-secondary {
```

Comment:
remove?

Reply:
I believe that the only use case we have for this is in facet search. Should we still remove? 
![image](https://user-images.githubusercontent.com/94572161/148613935-198d953a-9264-4db9-94b0-b099dcd42444.png)

Reply:
I wonder if the color distinction between parent and child is actually needed for this scenario. 
![image](https://user-images.githubusercontent.com/94572161/148622056-4c16b5fe-b78c-442f-a0f5-a017467a1e9b.png)

Reply:
I was done by design, so we should keep something similar. Perhaps we need to circle back with design regarding this? .checkbox-muted could be the color modifier? .checkbox could be the standard blue while -muted could be the new 'secondary'. Thoughts?

It's probably too much to hope that `border` or `background-color-medium` color would work, on all themes, any chance there?

Reply:
I think `border` ($palette-grey-30) and `body-background-medium` ($palette-grey-20) is too light with light theme and doesn't pass 3:1 color contrast ratio. 

I just realized that `$secondary` ($palette-grey-60) doesn't pass either when paired with the darker background in my screenshot above, even though it's what we have in prod right now.

The sweet spot seems to be `$palette-grey-80`, but that's currently unused by color/theme tokens. 

I updated to `$text-subtle` ($palette-grey-90) for now, which works with all themes. It's more like a `.checkbox-dark` rather than muted though.

Reply:
Design is good with the combo of $palette-grey-90 with a white checkmark.